extruct/rdfa.py

```diff
@@ -28,6 +29,104 @@
+            'rev'   : 'http://purl.org/stuff/rev#',
+            'rdfa'    : 'http://www.w3.org/ns/rdfa#',
+            'dc'    : 'http://purl.org/dc/terms/',
+            'foaf'    : 'http://xmlns.com/foaf/0.1/',
+            'void'    : 'http://rdfs.org/ns/void#',
+            'ical'    : 'http://www.w3.org/2002/12/cal/icaltzd#',
+            'vcard'   : 'http://www.w3.org/2006/vcard/ns#',
+            'wdrs'    : 'http://www.w3.org/2007/05/powder-s#',
+            'og'    : 'http://ogp.me/ns#',
+            'wdr'   : 'http://www.w3.org/2007/05/powder#',
+            'rdf'   : 'http://www.w3.org/1999/02/22-rdf-syntax-ns#',
+            'xhv'   : 'http://www.w3.org/1999/xhtml/vocab#',
+            'xsd'   : 'http://www.w3.org/2001/XMLSchema#',
+            'v'     : 'http://rdf.data-vocabulary.org/#',
+            'skosxl'  : 'http://www.w3.org/2008/05/skos-xl#',
+            'schema'  : 'http://schema.org/',
+        }
+
+        if 'http://' in prop:
+            return prop
+
+        prefix = prop.split(':')[0]
+        
+        match = None
+        if head_element.get('prefix'):
+            match = re.search(prefix + ': [^\s]+', head_element.get('prefix'))
+
+        # if namespace taken from prefix attribute in head tag
+        if match:
+            ns = match.group().split(': ')[1]
+            return ns + prop.split(':')[1]
+
+        # if namespace taken from xmlns attribute in html tag
+        if ('xmlns:' + prefix) in html_element.keys():
+            return html_element.get('xmlns:' + prefix) + prop.split(':')[1]
+
+        # if namespace present in inital context
+        if prefix in context:
+            return context[prefix] + prop.split(':')[1]
+
+        return prop
+    
+    # sorts the rdfa tags in jsonld string
+    def sort(self, json_object, ordered, key):
+        for i in range(len(ordered)):
+            if json_object[key][i]['@value'] != ordered[i]:
+                for j in range(i, len(json_object[key])):
+                    if json_object[key][j]['@value'] == ordered[i]:
+                        t = json_object[key][j]
+                        json_object[key][j] = json_object[key][i]
+                        json_object[key][i] = t
+                        continue
+    
+    
+    # fixes order of rdfa tags in jsonld string
+    # by comparing with order in document object
+    def fixOrder(self, jsonld_string, document):
+        try:
+            html_element = document.xpath('/html')[0]
+            head_element = document.xpath('//head')[0]
+        except IndexError:
+            return json.loads(jsonld_string)
+                
+        for meta_tag in head_element.xpath("meta[@property]"):
+            meta_tag.attrib['property'] = self.replaceNS(meta_tag.attrib['property'], html_element, head_element)
+            
+        json_objects = json.loads(jsonld_string)
+
+        for json_object in json_objects:
+            keys = json_object.keys()
+
+            ordered = []
```

Comment:
This is not required

Reply:
In all my tests, it always returned a list of size 1, with the first element in the list as the desired json. I was not sure about this though - will it always return a list of size 1?

Reply:
I don't get your point here :thinking: What I mean is that the variable `ordered` is initialized here as an empty list, but then this value is overwritten at line:
https://github.com/adityas114/extruct/blob/23425a0139db5973fb0d3af926d433c8375c393b/extruct/rdfa.py#L124
So it seems this initialized value is not used anywhere in any case.

Reply:
Sorry, I thought you were talking about the for loop. Okay I see the problem, I am more used to programming in Java 😄. I'll fix it.